packages/controls/src/serialization/function-serialization.ts

```diff
@@ -6,7 +6,16 @@ export function isFunction(value: any): value is AnyFunction {
   return typeof value === 'function'
 }
 
+type ResolveCallPromise<T extends AnyFunction> = (
+  value: SerializedFunctionReturnType<T>,
+) => void
+
+type OnMessageHandler<T extends AnyFunction> = MessagePort['onmessage'] & {
+  newCall?(resolve: ResolveCallPromise<T>): number
+}
```

Comment:
See below for why we are appending this optional method to the `onmessage` handler.